### ISSUE TITLE
1.6.0 Provider crashes during concurrent kibana object creation

### ISSUE THREAD

**Issue by timcosta** [phillbaker/terraform-provider-elasticsearch#197]
Heyo, wanted to pass along stack traces for a provider crash I just experienced:

```bash
Stack trace from the terraform-provider-elasticsearch_v1.6.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d3a497]

goroutine 117 [running]:
github.com/phillbaker/terraform-provider-elasticsearch/es.elastic7CreateIndexIfNotExists(0xc000752000, 0xc00015c120, 0x1c, 0xc00015c120, 0x1c, 0x1, 0x0, 0x0)
	github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_kibana_object.go:127 +0x257
github.com/phillbaker/terraform-provider-elasticsearch/es.resourceElasticsearchKibanaObjectCreate(0xc0007fb420, 0x1def6c0, 0xc00020e900, 0x2, 0x2aebdc0)
	github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_kibana_object.go:85 +0x18b
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0004f7680, 0xc0001e6320, 0xc0007ea420, 0x1def6c0, 0xc00020e900, 0x1eb5d01, 0xc000508428, 0xc0004f38f0)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/helper/schema/resource.go:310 +0x375
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000588400, 0xc0006d3a38, 0xc0001e6320, 0xc0007ea420, 0xc00020ebc8, 0xc00012c088, 0x1ebb180)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0000aeaa0, 0x2293f10, 0xc0004f2180, 0xc0007faf50, 0xc0000aeaa0, 0xc0004f2180, 0xc00088bba0)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1ff4500, 0xc0000aeaa0, 0x2293f10, 0xc0004f2180, 0xc0007661e0, 0x0, 0x2293f10, 0xc0004f2180, 0xc0007f2f00, 0x270)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc0007e6400, 0xc0006de420, 0x2aac6e0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.33.2/server.go:1210 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc0007e6400, 0x0)
	google.golang.org/grpc@v1.33.2/server.go:1533 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000ac210, 0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc0007e6400)
	google.golang.org/grpc@v1.33.2/server.go:871 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.33.2/server.go:869 +0x1fd
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d3a497]

goroutine 129 [running]:
github.com/phillbaker/terraform-provider-elasticsearch/es.elastic7CreateIndexIfNotExists(0xc00059a000, 0xc000676020, 0x1c, 0xc000676020, 0x1c, 0x1, 0x0, 0x0)
	github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_kibana_object.go:127 +0x257
github.com/phillbaker/terraform-provider-elasticsearch/es.resourceElasticsearchKibanaObjectCreate(0xc0003ff2d0, 0x1def6c0, 0xc00020e900, 0x2, 0x2aebdc0)
	github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_kibana_object.go:85 +0x18b
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0004f7680, 0xc0007ded70, 0xc000382060, 0x1def6c0, 0xc00020e900, 0x1eb5d01, 0xc0007425c8, 0xc0007570e0)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/helper/schema/resource.go:310 +0x375
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000588400, 0xc000649a38, 0xc0007ded70, 0xc000382060, 0xc000805048, 0xc000686168, 0x1ebb180)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0000aeaa0, 0x2293f10, 0xc000756030, 0xc0003fee00, 0xc0000aeaa0, 0xc000756030, 0xc00088fba0)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1ff4500, 0xc0000aeaa0, 0x2293f10, 0xc000756030, 0xc0007d0c60, 0x0, 0x2293f10, 0xc000756030, 0xc0003f8780, 0x274)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc000642200, 0xc0006de420, 0x2aac6e0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.33.2/server.go:1210 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc000642200, 0x0)
	google.golang.org/grpc@v1.33.2/server.go:1533 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000ac210, 0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc000642200)
	google.golang.org/grpc@v1.33.2/server.go:871 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.33.2/server.go:869 +0x1fd
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d3a497]

goroutine 21 [running]:
github.com/phillbaker/terraform-provider-elasticsearch/es.elastic7CreateIndexIfNotExists(0xc00059a1c0, 0xc0000c40e0, 0x1e, 0xc0000c40e0, 0x1e, 0x1, 0x0, 0x0)
	github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_kibana_object.go:127 +0x257
github.com/phillbaker/terraform-provider-elasticsearch/es.resourceElasticsearchKibanaObjectCreate(0xc000618a80, 0x1def6c0, 0xc00020e900, 0x2, 0x2aebdc0)
	github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_kibana_object.go:85 +0x18b
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0004f7680, 0xc000678190, 0xc00007e080, 0x1def6c0, 0xc00020e900, 0x1eb5d01, 0xc00019d7a8, 0xc00073de60)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/helper/schema/resource.go:310 +0x375
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000588400, 0xc00064da38, 0xc000678190, 0xc00007e080, 0xc0002bee08, 0xc0000ae8e8, 0x1ebb180)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0000aeaa0, 0x2293f10, 0xc00073d7a0, 0xc0006185b0, 0xc0000aeaa0, 0xc00073d7a0, 0xc000269ba0)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1ff4500, 0xc0000aeaa0, 0x2293f10, 0xc00073d7a0, 0xc00031e000, 0x0, 0x2293f10, 0xc00073d7a0, 0xc000616280, 0x26c)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc000642000, 0xc0006de420, 0x2aac6e0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.33.2/server.go:1210 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc000642000, 0x0)
	google.golang.org/grpc@v1.33.2/server.go:1533 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000ac210, 0xc0002dba40, 0x22a0058, 0xc000702a80, 0xc000642000)
	google.golang.org/grpc@v1.33.2/server.go:871 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.33.2/server.go:869 +0x1fd

Error: The terraform-provider-elasticsearch_v1.6.0 plugin crashed!
```

**Comment by timcosta** [comment]
If it helps, when i run with `-parallelism=1` i get a similar-ish issue:
```bash 
Stack trace from the terraform-provider-elasticsearch_v1.6.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d3a497]

goroutine 30 [running]:
github.com/phillbaker/terraform-provider-elasticsearch/es.elastic7CreateIndexIfNotExists(0xc0003d6000, 0xc0005504c0, 0x1c, 0xc0005504c0, 0x1c, 0x1, 0x0, 0x0)
	github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_kibana_object.go:127 +0x257
github.com/phillbaker/terraform-provider-elasticsearch/es.resourceElasticsearchKibanaObjectCreate(0xc0001f6e70, 0x1def6c0, 0xc0000dfd40, 0x2, 0x2aebdc0)
	github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_kibana_object.go:85 +0x18b
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000292870, 0xc0001f1720, 0xc00072bda0, 0x1def6c0, 0xc0000dfd40, 0x1eb5d01, 0xc000142f88, 0xc0002ef020)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/helper/schema/resource.go:310 +0x375
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000290400, 0xc00036fa38, 0xc0001f1720, 0xc00072bda0, 0xc0002e35e8, 0xc0000c15d0, 0x1ebb180)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000e848, 0x2293f10, 0xc0002ee7b0, 0xc0001f6770, 0xc00000e848, 0xc0002ee7b0, 0xc0006e9ba0)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1ff4500, 0xc00000e848, 0x2293f10, 0xc0002ee7b0, 0xc0001cf980, 0x0, 0x2293f10, 0xc0002ee7b0, 0xc0002f4280, 0x268)
	github.com/hashicorp/terraform-plugin-sdk@v1.13.1/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0007d4540, 0x22a0058, 0xc000297500, 0xc00013a400, 0xc000280fc0, 0x2aac6e0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.33.2/server.go:1210 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0007d4540, 0x22a0058, 0xc000297500, 0xc00013a400, 0x0)
	google.golang.org/grpc@v1.33.2/server.go:1533 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000282190, 0xc0007d4540, 0x22a0058, 0xc000297500, 0xc00013a400)
	google.golang.org/grpc@v1.33.2/server.go:871 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.33.2/server.go:869 +0x1fd

Error: The terraform-provider-elasticsearch_v1.6.0 plugin crashed!
```

and I realized I didn't provide my terraform, so here it is:
```hcl
resource "elasticsearch_kibana_object" "pattern" {
  for_each = toset(var.index_patterns)

  index = elasticsearch_opendistro_kibana_tenant.main.index
  body = <<EOF
[
  {
    "_id": "index-pattern:${each.value}",
    "_type": "doc",
    "_source": {
      "type": "index-pattern",
      "index-pattern": {
        "title": "${each.value}",
        "timeFieldName": "@timestamp"
      }
    }
  }
]
EOF
}```

**Comment by timcosta** [comment]
actually, i think this is related to #195 that I opened earlier, because if I do `index = lower(elasticsearch_opendistro_kibana_tenant.main.index)` the terraform applies cleanly.

**Comment by timcosta** [comment]
I believe this has been fixed in 1.6.1 or 1.6.2 so I am closing it.
